### PR TITLE
[26.x] Changes for rc3

### PR DESCRIPTION
**Pull request by fanquake**
Currently backports:
* https://github.com/bitcoin/bitcoin/pull/28858
* https://github.com/bitcoin/bitcoin/pull/28895 (partial)
* https://github.com/bitcoin/bitcoin/pull/28913
* https://github.com/bitcoin/bitcoin/pull/28905
* https://github.com/bitcoin/bitcoin/pull/28919
* https://github.com/bitcoin/bitcoin/pull/28925

Also includes changes for rc3, and reintegrating the release-notes.

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -1 +1,327 @@
...
+
+- RPC `walletprocesspsbt`, and `descriptorprocesspsbt` return
+  object now includes field `hex` (if the transaction
+  is complete) containing the serialized transaction
+  suitable for RPC `sendrawtransaction`. (#28414)
+
+- It's now possible to use [Miniscript](https://bitcoin.sipa.be/miniscript/) inside Taproot leaves for descriptor wallets. (#27255)
+
+GUI changes
+-----------
```
Comment (hebasto) [inline doc/release-notes.md:238]: I'm sorry for that late comment. Mind adding the following stuff:
```
- The transaction list in the GUI to no longer provide a special category for “payment to yourself”. Now transactions that have both inputs and outputs that affect the wallet are displayed on separate lines for spending and receiving. (gui#119)

- A new menu option allows migrating a legacy wallet based on keys and implied output script types stored in BerkeleyDB (BDB) to a modern wallet that uses descriptors stored in SQLite. (gui#738)

- The PSBT operations dialog marks outputs paying your own wallet with “own address”. (gui#740)

- The ability to create legacy wallets is being removed. (gui#764)
```
?

Comment (fanquake) [reply doc/release-notes.md:238]: Sure. Added to the next push.

**Comment by hebasto** [comment]
Suggesting to backport https://github.com/bitcoin/bitcoin/pull/28905 as well to ensure CI pass.

File: doc/release-notes.md [hunk 2]
```diff
@@ -1 +1,335 @@
...
+
+- The transaction list in the GUI to no longer provide a special category for "payment to yourself". Now transactions that have both inputs and outputs that affect the wallet are displayed on separate lines for spending and receiving. (gui#119)
```
Comment (hebasto) [inline doc/release-notes.md:240]: nit: 
```suggestion
- The transaction list in the GUI no longer provides a special category for "payment to yourself". Now transactions that have both inputs and outputs that affect the wallet are displayed on separate lines for spending and receiving. (gui#119)
```

I apologize for the wrong Grammar in my initial suggestion.

**Review by hebasto — APPROVED** [review]
ACK 740a85ce5f25f2282e32f148d61f92301bef7179, I've verified backports locally, also reviewed changes related to the `-par` option and `rc2` --> `rc3` as well.

File: doc/release-notes.md (hunk 2 above)
Comment (fanquake) [reply doc/release-notes.md:240]: No worries, fixed up.

**Review by hebasto — APPROVED** [review]
re-ACK f868852d3d7321aa522d6b168f33bef1e5eba2dc.

**Review by hebasto — APPROVED** [review]
re-ACK 2f86d3053314c382dfce5cf314e98811ed433859, only https://github.com/bitcoin/bitcoin/pull/28919 backported since my [recent](https://github.com/bitcoin/bitcoin/pull/28872#pullrequestreview-1744950215) review.

**Review by sedited — APPROVED** [review]
ACK 2f86d3053314c382dfce5cf314e98811ed433859

**Review by Mita23456 — COMMENTED** [review]
So not doing at

File: src/bench/pool.cpp [hunk 3]
```diff
@@ -37,8 +37,7 @@ static void PoolAllocator_StdUnorderedMapWithPoolResource(benchmark::Bench& benc
                                    std::hash<uint64_t>,
```
Comment (Mita23456) [inline src/bench/pool.cpp:37]: Ok

File: src/bench/pool.cpp [hunk 4]
```diff
@@ -37,8 +37,7 @@ static void PoolAllocator_StdUnorderedMapWithPoolResource(benchmark::Bench& benc
...
                                    std::equal_to<uint64_t>,
                                    PoolAllocator<std::pair<const uint64_t, uint64_t>,
-                                                 sizeof(std::pair<const uint64_t, uint64_t>) + 4 * sizeof(void*),
-                                                 alignof(void*)>>;
+                                                 sizeof(std::pair<const uint64_t, uint64_t>) + 4 * sizeof(void*)>>;
 
     // make sure the resource supports large enough pools to hold the node. We do this by adding the size of a few pointers to it.
     auto pool_resource = Map::allocator_type::ResourceType();
```
Comment (Mita23456) [inline src/bench/pool.cpp:43]: Yep

**Review by Mita23456 — COMMENTED** [review]
Hello

**Pull request merged**